src/plugins/fs/vfat.c

```diff
@@ -288,12 +288,26 @@ gboolean bd_fs_vfat_set_label (const gchar *device, const gchar *label, GError *
  * Tech category: always available
  */
 gboolean bd_fs_vfat_check_label (const gchar *label, GError **error) {
+    const gchar *forbidden = "\"*/:<>?\\|";
+    guint n;
+
     if (strlen (label) > 11) {
         g_set_error (error, BD_FS_ERROR, BD_FS_ERROR_LABEL_INVALID,
                      "Label for VFAT filesystem must be at most 11 characters long.");
         return FALSE;
     }
 
+    /* VFAT does not allow some characters; as dosfslabel does not enforce this,
+     * check in advance; also, VFAT only knows upper-case characters, dosfslabel
+     * enforces this */
```

Comment:
> VFAT only knows upper-case characters, dosfslabel enforces this

`dosfslabel` only warns about it. Maybe we should be stricter here and force uppercase only?

```
$ sudo dosfslabel /dev/loop1 aaaa
fatlabel: warning - lowercase labels might not work properly on some systems
```

Reply:
Right, though I'm not sure if it's worth enforcing that - i.e. failing the check. UDisks always converts FAT labels uppercase without complaining.

Pushed a second commit that always converts labels uppercase, please check. The `bd_fs_vfat_mkfs()` function still allows specifying arbitrary arguments, added a docs note.